src/client/scripts/esm/chess/variants/variant.ts

```diff
@@ -404,12 +510,16 @@ function getMovesetsOfVariant({ Variant, UTCDate = timeutil.getCurrentUTCDate(),
  * Any piece type present in the modifications will replace the default move that for that piece.
  * The slidelimit gamerule will only be applied to default movesets, not modified ones.
  * @param {Object} movesetModifications - The modifications to the default movesets.
- * @returns {number} [defaultSlideLimitForOldVariants] Optional. The slidelimit to use for default movesets, if applicable.
+ * @param {number} [defaultSlideLimitForOldVariants] Optional. The slidelimit to use for default movesets, if applicable.
  * @returns {Object} The pieceMovesets property of the gamefile.
  */
-function getMovesets(movesetModifications = {}, defaultSlideLimitForOldVariants) {
+function getMovesets(movesetModifications: Moveset = {}, defaultSlideLimitForOldVariants = Infinity) {
```

Comment:
defaultSlideLimitForOldVariants should be an optional argument, instead of defaulting to infinity, as movesets.getPieceDefaultMovesets() already has a default value for it when not specified.